src/PointwiseFunctions/GeneralRelativity/Ricci.cpp

```diff
@@ -5,35 +5,47 @@
 
 #include "DataStructures/Tensor/Tensor.hpp"
 #include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
 #include "Utilities/MakeWithValue.hpp"
 
 /// \cond
 namespace gr {
 template <size_t SpatialDim, typename Frame, IndexType Index, typename DataType>
-tnsr::aa<DataType, SpatialDim, Frame, Index> ricci_tensor(
+void ricci_tensor(
+    const gsl::not_null<tnsr::aa<DataType, SpatialDim, Frame, Index>*> result,
     const tnsr::Abb<DataType, SpatialDim, Frame, Index>& christoffel_2nd_kind,
     const tnsr::aBcc<DataType, SpatialDim, Frame, Index>&
         d_christoffel_2nd_kind) noexcept {
-  auto ricci = make_with_value<tnsr::aa<DataType, SpatialDim, Frame, Index>>(
+  *result = make_with_value<tnsr::aa<DataType, SpatialDim, Frame, Index>>(
```

Comment:
would `destructive_resize` work here? This current does a bunch of allocations even if the tensor is already allocated